api/hypershift/v1beta1/nodepool_conditions.go

```diff
@@ -18,6 +18,14 @@ const (
 	// NodePoolValidTuningConfigConditionType signals if the content within nodePool.spec.tuningConfig is valid.
 	// A failure here is unlikely to resolve without changing user input.
 	NodePoolValidTuningConfigConditionType = "ValidTuningConfig"
+	// NodePoolSupportedVersionSkewConditionType signals if the NodePool points to a version that falls within the supported skew policy with the HostedCluster.
+	// NodePool version cannot be higher than the HostedCluster version.
+	// For 4.y versions:
+	// - For 4.even versions (e.g. 4.18), allows up to 2 minor version differences (4.17, 4.16)
+	// - For 4.odd versions (e.g. 4.17), allows up to 1 minor version difference (4.16)
+	// When false, the NodePool will keep trying to operate as usual even though there are no guarantees
```

Comment:
nit: this line should grow a trailing period, in case consumers like https://pkg.go.dev/github.com/openshift/hypershift/api/hypershift/v1beta1#pkg-constants decide to start unwrapping Godocs for constants the way they already do for functions.